### PR TITLE
feat: Moving visualization rows from columns into filters [DHIS2-9106]

### PR DESCRIPTION
**Pull request by maikelarabori**
This PR adds a new Flyway script to move columns of type 'YEAR_OVER_YEAR_COLUMN' or 'YEAR_OVER_YEAR_LINE' and "dimension" 'dx' into filters.

Using a Java migration for this case. As recommended, we will let Hibernate deal with the sort_order column.

The reservation Pr for this Flyway is: https://github.com/dhis2/wow-backend/pull/38

### CODE REVIEW

**Comment by maikelarabori** [comment]
I found a problem in the logic. Becasue of sort_order we also have to "re-sort" the columns table. I'm going to add this fix.

**Comment by maikelarabori** [comment]
Hi @janhenrikoverland,
someone merged a Flyway migration (V2_35_10__Add_visualization_fontstyle.sql) without reservation. So my migration 2.35_10 is failing. I will have to rework on Monday to fix it.

Other than that the rest of checks should be all right.

Cheers,
Maikel

**Comment by maikelarabori** [comment]
@janhenrikoverland, actually my PR was kind of bypassed :), the reservation was done by Lars' PR: https://github.com/dhis2/dhis2-core/pull/5644.

I will have to re-create my PRs to fix it anyways. I hope Monday is not too late. Let me know if you need it for this weekend. Just drop me an e-mail. Thanks!

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v35/V2_35_11__Visualization_Move_Columns_To_Filters.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+            try (final ResultSet rs = ps.executeQuery())
+            {
+                rs.next();
+                return rs.getInt( "vis_counter" ) > 0;
+            }
+        }
+    }
+
+    private void insertIntoFilerTable( final Context context, final long visualizationId, final String dimension,
```
Comment (stian-sandvold) [inline dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v35/V2_35_11__Visualization_Move_Columns_To_Filters.java:121]: typo in method name, "Filer" not "Filter"

Comment (maikelarabori) [reply dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v35/V2_35_11__Visualization_Move_Columns_To_Filters.java]: Done

**Pull request merged**